src/ra_server.erl

```diff
@@ -352,24 +353,26 @@ init(#{id := Id,
     VotedFor = ra_log_meta:fetch(MetaName, UId, voted_for, undefined),
 
     LatestMacVer = ra_machine:version(Machine),
+    InitialMachineVersion = min(LatestMacVer,
+                                maps:get(initial_machine_version, Config, 0)),
```

Comment:
The client might be surprised by the fact that the effective version is lower than the requested `initial_machine_version`. It could lead to crashes or worse, difficult to detect non-desired behaviors.

Could we use the `machine_upgrade_strategy` here?
* `all`: all machine must support that `initial_machine_version` and a lower one is rejected, leading to a failure to init.
* `quorum`: a majority of machine needs to support the version, but not all of them. A lower supported version is accepted and the client is responsible for checking that’s the case when it determines the `initial_machine_version`.